tests/test-sources/plugins/lsp/language-servers/nil-ls.nix

```diff
@@ -1,5 +1,37 @@
 {
   example = {
+    plugins.lsp = {
+      enable = true;
+
+      servers.nil-ls = {
```

Comment:
ERROR: [lspconfig] Cannot access configuration for nil-ls. Ensure this server is listed in `server_configurations.md` or added as a custom server.

I guess we need an `upstreamName` or `lspconfigName` option (similar to `mkNeovimPlugin`'s `luaName`).

Reply:
Or, if we want our `plugins.lsp.servers.<name>`s to exactly match upstream's `server_configurations.md` names, perhaps we should drop this PR (until we have an upstream rename to deal with)?